### PR TITLE
fix `relabel` for permutation groups

### LINKED ISSUE #37379: Graph.relabel() misinterprets elements of .automorphism_group()
### Steps To Reproduce

Each element of automorphism group acts on the graph vertices by permuting (relabeling) them. It is therefore very natural for `.relabel()` to support automorphism group's elements. 

```
sage: G = graphs.CycleGraph(5)
sage: A = G.automorphism_group()
sage: H = G.random_orientation()
sage: a = A.random_element()
sage: print(a)
(0,3,1,4,2)
sage: H.relabel(a,inplace=False).show()
```

### Expected Behavior

For a given automorphism `a`, each graph vertex `v` should be relabeled into `a(v)`.

### Actual Behavior

At the moment `.relabel()` misunderstands a given graph's automorphism by issuing an wrongful error message `NotImplementedError: Non injective relabeling`. Elements of automorphism group **do** define injective labeling.

### Additional Information

A workaround is to call `H.relabel(lambda v: a(v),...)` instead of `H.relabel(a,...)`.

### Environment

```markdown
- **OS**: Ubuntu 22.04.3 LTS
- **Sage Version**: 10.3.beta7
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by dcoudert**
Fixes #37379.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by tscrim — APPROVED** [review]
LGTM.

File: src/sage/graphs/generic_graph.py [hunk 1]
```diff
@@ -23068,6 +23065,15 @@ def relabel(self, perm=None, inplace=True, return_map=False, check_input=True, c
             [0 0 1]
             [1 1 0]
 
+            sage: # needs sage.groups
+            sage: C5 = graphs.CycleGraph(5)
+            sage: C5.relabel({u: str(u) for u in C5}, inplace=True)
+            sage: ar = C5.automorphism_group().random_element()
```
Comment (tscrim) [inline src/sage/graphs/generic_graph.py:23071]: There is some trailing whitespace on 23071. Once fixed, you can revert to the positive review.

Comment (dcoudert) [reply src/sage/graphs/generic_graph.py:23071]: Done.

**Comment by dcoudert** [comment]
Thank you Travis for the review.

**Pull request merged**